firewall/privacy_mapper.go

```diff
@@ -812,6 +841,654 @@ func handleUpdatePolicyResponse(db firewalldb.PrivacyMapDB,
+					flags, randIntn, c.Channel.Capacity,
+				)
+				if err != nil {
+					return err
+				}
+
+				localBalance, err := maybeHideAmount(
+					flags, randIntn, c.Channel.LocalBalance,
+				)
+				if err != nil {
+					return err
+				}
+
+				remoteBalance, err := maybeHideAmount(
+					flags, randIntn, c.Channel.RemoteBalance,
+				)
+				if err != nil {
+					return err
+				}
+
+				chanPoint := c.Channel.ChannelPoint
+				if !flags.Contains(session.ClearChanIDs) {
+					chanPoint, err = firewalldb.HideChanPointStr(
+						tx, c.Channel.ChannelPoint,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				channel := lnrpc.PendingChannelsResponse_PendingOpenChannel{
+					// Non-obfuscated fields.
+					CommitFee:           c.CommitFee,
+					CommitWeight:        c.CommitWeight,
+					FeePerKw:            c.FeePerKw,
+					FundingExpiryBlocks: c.FundingExpiryBlocks,
+
+					Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+						// Obfuscated fields.
+						ChannelPoint:  chanPoint,
+						RemoteNodePub: remotePub,
+						Capacity:      capacity,
+						LocalBalance:  localBalance,
+						RemoteBalance: remoteBalance,
+
+						// Non-obfuscated fields.
+						ChanStatusFlags:       c.Channel.ChanStatusFlags,
+						Private:               c.Channel.Private,
+						CommitmentType:        c.Channel.CommitmentType,
+						Initiator:             c.Channel.Initiator,
+						LocalChanReserveSat:   c.Channel.LocalChanReserveSat,
+						RemoteChanReserveSat:  c.Channel.RemoteChanReserveSat,
+						NumForwardingPackages: c.Channel.NumForwardingPackages,
+						Memo:                  c.Channel.Memo,
+					},
+				}
+
+				pendingOpen[i] = &channel
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO: map other pending channel types.
+
+		return &lnrpc.PendingChannelsResponse{
+			TotalLimboBalance:           0,
+			PendingOpenChannels:         pendingOpen,
+			PendingClosingChannels:      []*lnrpc.PendingChannelsResponse_ClosedChannel{},
+			PendingForceClosingChannels: []*lnrpc.PendingChannelsResponse_ForceClosedChannel{},
+			WaitingCloseChannels:        []*lnrpc.PendingChannelsResponse_WaitingCloseChannel{},
+		}, nil
+	}
+}
+
+func handleClosedChannelsResponse(db firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	randIntn func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.ClosedChannelsResponse) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.ClosedChannelsResponse) (
+		proto.Message, error) {
+
+		closedChannels := make(
+			[]*lnrpc.ChannelCloseSummary,
+			len(r.Channels),
+		)
+
+		err := db.Update(func(tx firewalldb.PrivacyMapTx) error {
+			for i, c := range r.Channels {
+				var err error
+
+				remotePub := c.RemotePubkey
+				if !flags.Contains(session.ClearPubkeys) {
+					remotePub, err = firewalldb.HideString(
+						tx, remotePub,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				capacity, err := maybeHideAmount(
```

Comment:
How come we obfuscate the capacity here and in the Pending Channels response, but we don't obfuscate the capacity in the `listchannels` response? I think the latter is related to the `AutoFees` algo right? Given that though, should we mimic that behaviour here and for the Pending Channels response as well?

Reply:
> I think the latter is related to the AutoFees algo right?
Yes, that's right, we need more work to study the influence on the algo, but the goal would be to obfuscate there as well.